### ISSUE TITLE
Azure Managed Identity Auth doesn't work in linux consumption plan

### ISSUE THREAD

**Issue by older** [dotnet/SqlClient#863]
### Describe the bug
Using the "2017-09-01" Azure Managed Identity API version [is required on Azure Functions Linux Consumption hosting plans](https://docs.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=dotnet#using-the-rest-protocol).  Current implementation uses [version 2019-08-01](https://github.com/dotnet/SqlClient/blob/6a6a9df90b71d7723cc6a74295d8d4c180ac3804/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/AzureManagedIdentityAuthenticationProvider.cs#L22) which produces the following exception:

```
Exception message:
Received a non-retryable error. Identity Response Code: BadRequest, Response: {"error":{"code":"UnsupportedApiVersion","message":"The HTTP resource that matches the request URI 'http://localhost:8081/msi/token?resource=https://database.windows.net/&api-version=2019-08-01' does not support the API version '2019-08-01'.","innerError":null}} 
Stack trace:
System.AggregateException:
   at Microsoft.Data.ProviderBase.DbConnectionPool.CheckPoolBlockingPeriod (Microsoft.Data.SqlClient, Version=2.0.20168.4, Culture=neutral, PublicKeyToken=23ec7fc2d6eaa4a5)
   at Microsoft.Data.ProviderBase.DbConnectionPool.CreateObject (Microsoft.Data.SqlClient, Version=2.0.20168.4, Culture=neutral, PublicKeyToken=23ec7fc2d6eaa4a5)
   at Microsoft.Data.ProviderBase.DbConnectionPool.UserCreateRequest (Microsoft.Data.SqlClient, Version=2.0.20168.4, Culture=neutral, PublicKeyToken=23ec7fc2d6eaa4a5)
   at Microsoft.Data.ProviderBase.DbConnectionPool.TryGetConnection (Microsoft.Data.SqlClient, Version=2.0.20168.4, Culture=neutral, PublicKeyToken=23ec7fc2d6eaa4a5)
   at Microsoft.Data.ProviderBase.DbConnectionPool.WaitForPendingOpen (Microsoft.Data.SqlClient, Version=2.0.20168.4, Culture=neutral, PublicKeyToken=23ec7fc2d6eaa4a5)
...
```

### To reproduce
Include a complete code listing (or project/solution) that we can run to reproduce the issue.

Partial code listings, or multiple fragments of code, will slow down our response or cause us to push the issue back to you to provide code to reproduce the issue.

```c#
string ConnectionString = @"Server=demo.database.windows.net; Authentication=Active Directory Managed Identity; Database=testdb;";
using (SqlConnection conn = new SqlConnection(ConnectionString)) {
    conn.Open();
}
```

### Expected behavior
Database connection is open.

### Further technical details
Microsoft.Data.SqlClient version: 2.1.1
.NET target: Core 3.1.10
SQL Server version: Azure SQL
Operating system: Azure Functions Linux Consumption hosting plan

**Comment by cheenamalhotra** [comment]
Acknowledged.
We'll consider addressing this in next iterations.

**Comment by older** [comment]
@cheenamalhotra Maybe [documentation](https://docs.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication#using-active-directory-managed-identity-authentication) should be updated to mention this?

May I ask why [Azure.Identity](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity) was not used to implement this? Was it too heavy a dependency for SqlClient?

**Comment by cheenamalhotra** [comment]
Hi @older,

There's an open issue for Azure Identity dependency work we'll be doing soon (#771) 

To summarize, it wasn't possible in the last release as Azure Identity doesn't support `net46` and dropping `net46` support wasn't possible in release v2.1.x for SqlClient.

**Comment by cheenamalhotra** [comment]
This has been fixed with the latest release [v3.0.0-preview2](https://www.nuget.org/packages/Microsoft.Data.SqlClient/3.0.0-preview2.21106.5). The driver now depends on Azure.Identity to acquire access token for Managed Identity based authentication. [Release notes](https://github.com/dotnet/SqlClient/blob/main/release-notes/3.0/3.0.0-preview2.md#microsoftdatasqlclient-300-preview2211065-released-16-april-2021)

Closing issue as resolved.
